src/libs/OptionsListUtils.ts

```diff
@@ -543,24 +544,22 @@ function getAllReportErrors(report: OnyxEntry<Report>, reportActions: OnyxEntry<
         if (ReportUtils.shouldShowRBRForMissingSmartscanFields(report?.reportID ?? '-1') && !ReportUtils.isSettled(report?.reportID)) {
             reportActionErrors.smartscan = ErrorUtils.getMicroSecondOnyxErrorWithTranslationKey('iou.error.genericSmartscanFailureMessage');
         }
-    } else if (ReportUtils.hasSmartscanError(Object.values(reportActions ?? {}))) {
+    } else if (ReportUtils.hasSmartscanError(reportActionsArray)) {
         reportActionErrors.smartscan = ErrorUtils.getMicroSecondOnyxErrorWithTranslationKey('iou.error.genericSmartscanFailureMessage');
     }
     // All error objects related to the report. Each object in the sources contains error messages keyed by microtime
-    const errorSources = {
-        reportErrors,
-        ...reportErrorFields,
-        ...reportActionErrors,
-    };
-    // Combine all error messages keyed by microtime into one object
-    const allReportErrors = Object.values(errorSources)?.reduce((prevReportErrors, errors) => {
-        if (isEmptyObject(errors)) {
-            return prevReportErrors;
-        }
+    // eslint-disable-next-line prefer-object-spread
```

Comment:
Can we update the code to remove the need for this suppression?

Reply:
The change was made for performance reasons since it's  faster and uses fewer resources than object spread syntax. However, ESLint prefers object spread over `Object.assign()`, leading to the need for the suppression comment to keep the performance benefits without  changing linting rules.

Reply:
Should we consider removing this spread preference that is enforced by default?

Reply:
>The change was made for performance reasons since it's faster and uses fewer resources than object spread syntax. However, ESLint prefers object spread over Object.assign(), leading to the need for the suppression comment to keep the performance benefits without changing linting rules.

Ah yes that makes sense, thank you for the explanation. Can you please add this as a comment in the code? 

> Should we consider removing this spread preference that is enforced by default?

I have no opinion on that. If you think it's a good idea, perhaps you can raise it in #expensify-open-source ?

Reply:
Commented.  Will also raise the removal of this spread preference as a proposal and create a separate PR for that.